src/dialog.js

```diff
@@ -452,6 +464,146 @@ genotet.dialog.upload_ = function() {
 };
 
 /**
+ * Creates a dialog for signing up.
+ * @private
+ */
+genotet.dialog.signUp_ = function() {
+  var modal = $('#dialog');
+  modal.find('.modal-content').load(genotet.dialog.TEMPLATES_.signUp,
+    function() {
+      modal.modal();
+      var email = modal.find('#email');
+      var username = modal.find('#username');
+      var password = modal.find('#password');
+      var confirmPassword = modal.find('#confirm-password');
+      var btnSignUp = modal.find('#btn-sign-up').prop('disabled', true);
+      var inputValid = false;
+
+      var uploadReady = function() {
+        // Validate the input information.
+        inputValid = genotet.utils.validateEmail(
+            /** @type {string} */(email.val())) &&
+          genotet.utils.validateUsername(
+            /** @type {string} */(username.val())) &&
+          genotet.utils.validatePassword(
+            /** @type {string} */(password.val()));
+
+        // Checks if all required fields are filled.
+        return email.val() !== '' && username.val() !== '' &&
+          password.val() !== '' && (confirmPassword.val() == password.val() &&
+          inputValid);
+      };
+      confirmPassword.on('input', function() {
+        btnSignUp.prop('disabled', !uploadReady());
+      });
+      btnSignUp.click(function() {
+        var userInfo = {
+          type: 'sign-up',
+          email: email.val(),
+          username: username.val(),
+          password: CryptoJS.SHA256(/** @type {string} */(password.val()))
```

Comment:
under https you can send plaintext string. Then you can validate password complexity on the server.